cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -293,6 +303,41 @@ func (o *objectGraph) getDiscoveryTypes() error {
 	return nil
 }
 
+// getDiscoveryFileTypes returns the list of TypeMeta to be considered for a save / restore cycle.
+// This list includes all the types defined by a slice of objects populated from file
+func (o *objectGraph) getDiscoveryFileTypes(objs []unstructured.Unstructured) error {
```

Comment:
I'm not sure that resuming type informations for object instances is the right approach, because we are missing all the information that comes from annotations at CRD level, which are getting more and more relevant in the move logic.
As commented above, let's try to use getDiscoveryTypes reading from CRD.